Source/Core/Core/FifoPlayer/FifoRecorder.cpp

```diff
@@ -6,13 +6,168 @@
 #include <algorithm>
 #include <cstring>
 
+#include "Common/Logging/Log.h"
 #include "Common/MsgHandler.h"
 #include "Common/Thread.h"
+
 #include "Core/ConfigManager.h"
-#include "Core/FifoPlayer/FifoAnalyzer.h"
-#include "Core/FifoPlayer/FifoRecordAnalyzer.h"
 #include "Core/HW/Memmap.h"
 
+#include "VideoCommon/OpcodeDecoding.h"
+#include "VideoCommon/XFStructs.h"
+
+class FifoRecorder::FifoRecordAnalyzer : public OpcodeDecoder::Callback
+{
+public:
+  explicit FifoRecordAnalyzer(FifoRecorder* owner) : m_owner(owner) {}
+  explicit FifoRecordAnalyzer(FifoRecorder* owner, const u32* cpmem)
+      : m_owner(owner), m_cpmem(cpmem)
+  {
+  }
+
+  OPCODE_CALLBACK(void OnXF(u16 address, u8 count, const u8* data)) {}
+  OPCODE_CALLBACK(void OnCP(u8 command, u32 value)) { GetCPState().LoadCPReg(command, value); }
+  OPCODE_CALLBACK(void OnBP(u8 command, u32 value)) {}
+  OPCODE_CALLBACK(void OnIndexedLoad(CPArray array, u32 index, u16 address, u8 size));
+  OPCODE_CALLBACK(void OnPrimitiveCommand(OpcodeDecoder::Primitive primitive, u8 vat,
+                                          u32 vertex_size, u16 num_vertices,
+                                          const u8* vertex_data));
+  OPCODE_CALLBACK(void OnDisplayList(u32 address, u32 size))
+  {
+    WARN_LOG_FMT(VIDEO,
+                 "Unhandled display list call {:08x} {:08x}; should have been inlined earlier",
+                 address, size);
+  }
+  OPCODE_CALLBACK(void OnNop(u32 count)) {}
+  OPCODE_CALLBACK(void OnUnknown(u8 opcode, const u8* data)) {}
+
+  OPCODE_CALLBACK(void OnCommand(const u8* data, u32 size)) {}
+
+  OPCODE_CALLBACK(CPState& GetCPState()) { return m_cpmem; }
+
+private:
+  void ProcessVertexComponent(CPArray array_index, VertexComponentFormat array_type,
+                              u32 component_offset, u32 vertex_size, u16 num_vertices,
+                              const u8* vertex_data);
+
+  const FifoRecorder* m_owner;
```

Comment:
This field is currently unused (the code that should use it instead uses `FifoRecorder::GetInstance()`).